alerter.c

```diff
@@ -1,19 +1,34 @@
 #include <stdio.h>
 #include <assert.h>
+#include "config.h"
+#include "alerter.h"
 
+//Variable to holder failure count on alert
 int alertFailureCount = 0;
 
+#ifdef STUB

```

Comment:
Using conditional compilation for separating production and test code may get complicated as the code size grows.
it is better to use interfaces or functions pointers or lambda functions (if available) to inject either required production or test code based on the envt.
Production code should be simple in the sense, given an input the output should be deterministic, may it be dev environment or test or field

Reply:
Updated review comments.